### PR TITLE
fix: set JSONRenderer as the DEFAULT_RENDERER_CLASSES

### PR DESCRIPTION
**Pull request by CodeWithEmad**
This will introduce a similar approach as `edx-platform` to set `JSONRenderer` as the `DEFAULT_RENDERER_CLASSES`. with this, we don't have Browsable APIs anymore. also, some code reformatting with black and RST cleanup in README. 

See here for more context: https://github.com/overhangio/tutor-notes/pull/35

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @CodeWithEmad! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by kdmccormick** [comment]
Thanks for the PR @CodeWithEmad . I think the JSON change looks good. Let me see if there is a maintainer that'd want to review this before I merge it.

Regarding black, there is a cost associated with reformatting, in that it complicates git blame and makes it harder for folks with forks to rebase. So, my rule of thumb is that we should only reformat it if we also enforce the new style in CI so that we never need to apply the same reformatting again. Please remove the black commit from this PR, but feel free to introduce it in a new PR along with a black CI check if that interests you.

(the docs reformatting is good to stay -- the considerations about git and forks do not apply so much :)

**Comment by CodeWithEmad** [comment]
Sure @kdmccormick. Thanks for the explaination.

**Comment by e0d** [comment]
@CodeWithEmad I've run the test.  I've also added you to the triage team as you are covered under an entity CLA.  This means that tests should run automatically for you in the future.  Let me know if that doesn't happen.

**Review by kdmccormick — APPROVED** [review]
Looks good!

Let's see if re-running those migration checks fixes them...

**Comment by mphilbrick211** [comment]
@CodeWithEmad the tests have been re-run - looks like some checks are still failing.

**Comment by CodeWithEmad** [comment]
That's super odd. I've added 1 DRF config and the tests are failing on installing the dependencies! I'll look into it.

**Comment by CodeWithEmad** [comment]
@mphilbrick211 All checks have passed.

**Comment by openedx-webhooks** [comment]
@CodeWithEmad 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-notes-api&entry.867055334=https://github.com/openedx/edx-notes-api/pull/403&entry.1484655318=https://github.com/CodeWithEmad&entry.752974735=2024-04-22+15:21&entry.1917517419=2024-05-06+13:23&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by kdmccormick** [comment]
Thanks @CodeWithEmad !

**Pull request merged**
